templates/zerver/emails/email.css

```diff
@@ -3,418 +3,425 @@
 because hex color codes are rendered by all email clients
 consistently.  Whereas HSL colors don't work with Hotmail. */
 
-img {
-    border: none;
-    -ms-interpolation-mode: bicubic;
-    max-width: 100%;
-    vertical-align: bottom;
-}
-
-body {
-    font-family: sans-serif;
-    font-size: 14px;
-    line-height: 1.4;
-    margin: 0;
-    padding: 0;
-    -ms-text-size-adjust: 100%;
-    -webkit-text-size-adjust: 100%;
-}
-
-pre {
-    font-family: sans-serif;
-    font-size: 14px;
-}
-
-table {
-    border-collapse: separate;
-    mso-table-lspace: 0;
-    mso-table-rspace: 0;
-    width: 100%;
-}
-
-table td {
-    font-family: sans-serif;
-    font-size: 14px;
-    vertical-align: top;
-}
-
-.body {
-    width: 100%;
-}
-
-.container {
-    display: block;
-    margin: 0 auto !important;
-    max-width: 500px;
-    padding: 10px;
-}
-
-.container.wide-container {
-    max-width: 700px;
-}
-
-.content {
-    box-sizing: border-box;
-    display: block;
-    margin: 20px auto 0;
-    max-width: 580px;
-    padding: 10px;
-}
-
-.content.wide-content {
-    max-width: 780px;
-}
-
-.main {
-    background-color: #fff;
-    border-radius: 3px;
-    width: 100%;
-}
-
-.wrapper {
-    box-sizing: border-box;
-    padding: 0 20px 20px;
-}
-
-.footer {
-    clear: both;
-    padding-top: 10px;
-    text-align: center;
-    width: 100%;
-    margin-top: 20px;
-}
+/* This is a workaround to avoid applying styles to dynamically
+   created content, such as in `missed_message.html`. Our goal is
+   to eliminate the use of the .no_inlined_css_inside class name
+   and instead apply all of the styles from this file to the HTML
+   of the emails. */
+:not(.no_inlined_css_inside) {
```

Comment:
This is not logically correct. `:not(.no_inlined_css_inside) img` matches “`<img>` with some ancestor lacking the `no_inlined_css_inside` class”, not “`<img>` with no ancestor having the `no_inlined_css_inside` class”.

If not all elements should get the below styling, we need to find a way to specify which elements *should* get it, not which elements *should not* get it.